jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -680,40 +680,31 @@ ecma_op_object_find_own (ecma_value_t base_value, /**< base value */
  *         Returned value must be freed with ecma_free_value
  */
 ecma_value_t
-ecma_op_object_find_by_uint32_index (ecma_object_t *object_p, /**< the object */
-                                     uint32_t index) /**< property index */
+ecma_op_object_find_by_index (ecma_object_t *object_p, /**< the object */
+                              ecma_length_t index) /**< property index */
 {
   if (JERRY_LIKELY (index <= ECMA_DIRECT_STRING_MAX_IMM))
   {
     return ecma_op_object_find (object_p, ECMA_CREATE_DIRECT_UINT32_STRING (index));
   }
 
-  ecma_string_t *index_str_p = ecma_new_non_direct_string_from_uint32 (index);
-  ecma_value_t ret_value = ecma_op_object_find (object_p, index_str_p);
-  ecma_deref_ecma_string (index_str_p);
-
-  return ret_value;
-} /* ecma_op_object_find_by_uint32_index */
+  ecma_string_t *index_str_p;
+#if ENABLED (JERRY_ESNEXT)
+  if (JERRY_UNLIKELY (index > UINT32_MAX))
```

Comment:
Are these functions that much of a bottleneck to warrant manually inlining these? Just curious, to me it seems the performance cost of calling for example `ecma_new_ecma_string_from_length` would be negligible compared to the find operation itself.